tools/gyp/pylib/gyp/generator/msvs.py

```diff
@@ -3036,7 +3036,7 @@ def _GetMSBuildLocalProperties(msbuild_toolset):
             [
                 "PropertyGroup",
                 {"Label": "Locals"},
-                ["PlatformToolset", msbuild_toolset],
+                ["PlatformToolset", "ClangCL"],
```

Comment:
Let's start with this one. We need to go back to `msbuild_toolset` and find a way to set its value to `ClangCL` conditionally from `node.gyp` or `node.gypi`. Then make a PR to https://github.com/nodejs/gyp-next with the changes in `tools/gyp`

Reply:
No need to change gyp: https://github.com/nodejs/node/pull/52870/commits/9c0ad10a7d5a32ea232689c7880c821adcd6fd6a